packages/core/src/render3/instructions/di.ts

```diff
@@ -49,6 +51,16 @@ export function ɵɵdirectiveInject<T>(token: ProviderToken<T>, flags = InjectFl
     return ɵɵinject(token, flags);
   }
   const tNode = getCurrentTNode();
+
+  if (typeof ngDevMode === 'undefined' || ngDevMode) {
+    const value = getOrCreateInjectable<T>(tNode as any, lView, resolveForwardRef(token), flags);
+    injectorProfiler({
+      type: InjectorProfilerEventType.Inject,
+      data: {token: token as Type<unknown>, value: value as Type<unknown>, flags}
+    });
+    return value;
+  }
+
   return getOrCreateInjectable<T>(
       tNode as TDirectiveHostNode, lView, resolveForwardRef(token), flags);
```

Comment:
Extract the code that gets the value outside of the `if` block:
```suggestion
  const value = getOrCreateInjectable<T>(tNode as any, lView, resolveForwardRef(token), flags);

  if (typeof ngDevMode === 'undefined' || ngDevMode) {
    injectorProfiler({
      type: InjectorProfilerEventType.Inject,
      data: {token: token as Type<unknown>, value: value as Type<unknown>, flags}
    });
    return value;
  }

  return value;
```